### PR TITLE
Create InputField component

### PR DESCRIPTION
**Pull request by KGhising**
Implement Input type toggle with icon

### CODE REVIEW

**Comment by KGhising** [comment]
@silwalanish should build different component for password toggle?

**Comment by silwalanish** [comment]
> @silwalanish should build different component for password toggle?

Sure.

File: src/Components/InputField.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+import '../Themes/inputfield.css';
```
Comment (silwalanish) [inline src/Components/InputField.jsx:4]: Fixes the build  error.
```suggestion
import '../Themes/InputField.css';
```

File: src/Components/Icon.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  faArrowRight,
+  faArrowUp,
+  faArrowDown,
+  faChartBar,
+  faArrowAltCircleRight,
+  faArrowAltCircleLeft,
+  faArrowAltCircleUp,
+  faArrowAltCircleDown,
+  faHeart
+);
```
Comment (silwalanish) [inline src/Components/Icon.jsx:50]: Create a file `iconSettings.js` in `src` folder and move these there.

File: src/App.jsx [hunk 3]
```diff
@@ -20,18 +20,20 @@ export default class App extends React.Component {
...
-            <Route path={DASHBOARD_URL}>
-              <Dashboard></Dashboard>
-            </Route>
-            <Route path={HOME_URL}>
-              <Home></Home>
-            </Route>
-          </Switch>
-        </BrowserRouter>
-      </Provider>
+      <div>
```
Comment (silwalanish) [inline src/App.jsx:23]: Why do we need this `div`?

File: src/Components/InputField.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+  render() {
+    return (
+      <div>
+        <i className='icon-left'>
+          <Icon size='xs' name={this.props.leftIconName} />
+        </i>
+        <i className='icon-right' onClick={this.togglePasswordVisiblity}>
+          <Icon size='xs' name={this.state.changeIcon ? 'eye-slash' : 'eye'} />
+        </i>
```
Comment (silwalanish) [inline src/Components/InputField.jsx:46]: Move the after the `input` field.

File: src/Components/InputField.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      this.setState({ password: this.props.password });
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        <i className='icon-left'>
+          <Icon size='xs' name={this.props.leftIconName} />
+        </i>
```
Comment (silwalanish) [inline src/Components/InputField.jsx:43]: This should be wrapped with a button if there is a `onLeftIconClick` prop.

File: src/Components/InputField.jsx (hunk 4 above)
Comment (silwalanish) [reply src/Components/InputField.jsx:46]: This should also be in a button if `onRightIconClick` prop exists.

Comment (silwalanish) [inline src/Components/InputField.jsx:46]:
```suggestion
        { this.getRightIcon() }
```
```jsx
getRightIcon() {
  return this.getIconButton(this.props.rightIcon, this.props.onRightIconClick, 'icon-right');
}

getLeftIcon() {
  return this.getIconButton(this.props.leftIcon, this.props.onLeftIconClick, 'icon-left');
}

getIconButton(icon, onClickHandle, iconClass) {
  return (icon &&
    (onClickHandle)
    ?
      <button type='button' onClick={onClickHandle}>
        { this.getIcon(iconClass, icon) }
      </button>
    :
      this.getIcon(iconClass, icon)
  );
}

getIcon(clsName, iconName) {
 return (
    <i className={clsName}>
      <Icon size='xs' name={iconName} />
    </i>
 );
}
```

File: src/Components/InputField.jsx (hunk 5 above)
Comment (silwalanish) [reply src/Components/InputField.jsx:43]:
```suggestion
        { this.getLeftIcon() }
```

**Pull request merged**